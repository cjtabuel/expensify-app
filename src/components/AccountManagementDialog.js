import React, { useState } from 'react'

import {
  Button,
  Checkbox,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  FormControlLabel,
  Grid,
} from '@material-ui/core';

const AccountManagementDialog = ({ accountManagementDialogOpen, setAccountManagementDialogOpen, handleDeleteUserAccount }) => {
  const [confirmDeletion, setConfirmDeletionState] = useState(false)
  const [confirmUnderstanding, setConfirmUnderstandingState] = useState(false)

  const handleDeleteUser = () => {
    if (!!confirmDeletion && !!confirmUnderstanding) {
      handleDeleteUserAccount()
    }
  }
  const handleClose = () => {
    setAccountManagementDialogOpen(false)
    setConfirmDeletionState(false)
    setConfirmUnderstandingState(false)
  }
  const handleConfirmDeletion = (e) => {
    setConfirmDeletionState(e.target.checked)
  }
  const handleConfirmUnderstanding = (e) => {
    setConfirmUnderstandingState(e.target.checked)
  }

  return (
    <div>
      <Dialog open={accountManagementDialogOpen} onClose={handleClose} aria-labelledby="form-dialog-title">
        <DialogTitle id="form-dialog-title">Delete Account</DialogTitle>
        <DialogContent dividers>
          <Grid container direction='column' spacing={2}>
            <Grid item xs={12}>
              <DialogContentText>
                &#42; Are you sure you want to delete your account?
              </DialogContentText>
              <FormControlLabel
                control={
                  <Checkbox
                    checked={confirmDeletion}
                    onChange={handleConfirmDeletion}
                    inputProps={{ 'aria-label': 'primary checkbox' }}
                  />}
                label='Yes, I am sure.'
              />
            </Grid>
            <Grid item xs={12}>
              <DialogContentText>
                &#42; Deleting your account would mean the deletion of your complete account with Expensify and all your recorded expenses.
              </DialogContentText>
              <DialogContentText>
                &#42; If you choose to continue, we cannot retrieve your data should you choose to join us again in the future.
              </DialogContentText>
              <FormControlLabel
                control={
                  <Checkbox
                    checked={confirmUnderstanding}
                    onChange={handleConfirmUnderstanding}
                    inputProps={{ 'aria-label': 'primary checkbox' }}
                  />}
                label='I understand.'
              />
            </Grid>
          </Grid>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={handleDeleteUser}
            color="primary"
            variant='contained'
            disabled={!!confirmDeletion && !!confirmUnderstanding ? false : true}
          >
            Confirm
          </Button>
          <Button onClick={handleClose} color="secondary" variant='contained'>
            Cancel
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}

export default AccountManagementDialog
