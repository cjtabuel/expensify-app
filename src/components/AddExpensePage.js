import React, { useCallback } from 'react'
import ExpenseForm2 from './ExpenseForm2'
import { useDispatch } from 'react-redux'
import { startAddExpense } from '../actions/expenses'

import {
  Box,
  Button,
  Container,
  Grid,
  Paper,
  Typography
} from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    '& > *': {
      paddingTop: theme.spacing(3),
      marginBottom: theme.spacing(6),
      width: theme.spacing(700),
      height: theme.spacing(14),
      backgroundColor: theme.palette.type === 'dark' ? '#2a2a2a' : '#f7f7f7',
    }
  },
}));

const AddExpensePage = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch()
  const startAddExpenseAction = useCallback((expense) => dispatch(startAddExpense(expense)))

  const onSubmit = (expense) => {
    startAddExpenseAction(expense)
    props.history.push('/')
  }

  return (
    <div>
      <div className={classes.root}>
        <Paper elevation={0} square>
          <Container fixed>
            <Typography variant="h4" align="left">
              <Box fontWeight={300} m={1}>
                Add Expense
              </Box>
            </Typography>
          </Container>
        </Paper>
      </div>
      <ExpenseForm2
        onSubmit={onSubmit}
      />
    </div>
  )
}

export { AddExpensePage as default }