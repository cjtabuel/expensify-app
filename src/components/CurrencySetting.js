import React, { useState, useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSetCurrencySetting } from '../actions/app-settings'

import {
  Box,
  Button,
  TextField,
  MenuItem,
  Grid,
  ListItemIcon,
  Typography,
  Snackbar,
} from '@material-ui/core'

import MuiAlert from '@material-ui/lab/Alert';

import currencyFixtures from '../fixtures/currencyFixtures'
import SettingsIcon from '@material-ui/icons/Settings';
import SaveIcon from '@material-ui/icons/Save'

const Alert = (props) => {
  return <MuiAlert elevation={6} variant="filled" {...props} />;
}

const CurrencySetting = () => {
  // dispatch actions
  const dispatch = useDispatch()
  const setCurrencySettingAction = useCallback((currency) => dispatch(startSetCurrencySetting(currency)))

  // fetch user currency setting
  const currencySettings = useSelector((state) => state.appSettings.currencySettings)

  // set user currency preferences
  const [currency, setCurrency] = useState(currencySettings)

  // set save success msg
  const [saveSuccess, setSaveSucccessMsg] = useState(false)

  const handleCurrencySetting = () => {
    setCurrencySettingAction(currency)
      .then(() => {
        setSaveSucccessMsg(true)
      })
  }
  const handleChange = (e) => {
    const currencyOptions = currencyFixtures.find((option) => option.value === e.target.value)
    setCurrency(currencyOptions)
  }
  const handleCloseSaveSuccessMsg = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    setSaveSucccessMsg(false);
  };

  return (
    <div>
      <Grid container direction="row" alignItems="center" justify="flex-start" spacing={1}>
        <Grid item xs={1}>
          <ListItemIcon>
            <SettingsIcon />
          </ListItemIcon>
        </Grid>
        <Grid item>
          <Typography variant="h6" align="left">
            <Box fontWeight={300} m={1}>
              Application Settings
              </Box>
          </Typography>
        </Grid>
      </Grid>
      <Grid container spacing={2} direction="row">
        <Grid item xs={6}>
          <div>
            <TextField
              id="outlined-select-currency"
              select
              label="Select"
              value={currency.value}
              onChange={handleChange}
              helperText="Please select your currency"
              variant="outlined"
            >
              {currencyFixtures.map((option) => (
                <MenuItem key={option.value} value={option.value}>
                  {option.label} {option.value} <span>123.45</span>
                </MenuItem>
              ))}
            </TextField>
          </div>
        </Grid>

        <Grid item xs={6}>
          <Typography variant="caption" align="center">
            &#8226; This setting will be applied as the currency for your expenses
        </Typography>
        </Grid>
        <Grid item xs={12}>
          <Button
            size="medium"
            variant="contained"
            color="primary"
            startIcon={<SaveIcon />}
            onClick={handleCurrencySetting}
          >
            Save
          </Button>
        </Grid>
        <Grid item xs={12}>
          <hr />
        </Grid>
      </Grid>
      <Snackbar
        open={saveSuccess}
        autoHideDuration={6000}
        onClose={handleCloseSaveSuccessMsg}
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        >
        <Alert onClose={handleCloseSaveSuccessMsg} severity="success">
          Your currency setting has been updated successfully!
        </Alert>
      </Snackbar>
    </div>
  )
}

export default CurrencySetting