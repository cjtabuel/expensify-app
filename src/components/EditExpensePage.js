import React, { useState, useCallback } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import ExpenseForm2 from './ExpenseForm2'
import RemoveExpenseDialog from './RemoveExpenseDialog'
import { startRemoveExpense } from '../actions/expenses'
import { startEditExpense } from '../actions/expenses'

import {
  Box,
  Button,
  Container,
  Divider,
  Grid,
  ListItemIcon,
  Paper,
  Typography
} from '@material-ui/core'
import DeleteIcon from '@material-ui/icons/Delete';
import SettingsIcon from '@material-ui/icons/Settings';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    '& > *': {
      paddingTop: theme.spacing(3),
      marginBottom: theme.spacing(6),
      width: theme.spacing(700),
      height: theme.spacing(14),
      backgroundColor: theme.palette.type === 'dark' ? '#2a2a2a' : '#f7f7f7',
    },
  },
}));

const EditExpensePage = props => {
  const classes = useStyles();
  const dispatch = useDispatch()

  // edit and remove expense dispatch actions
  const startEditExpenseAction = useCallback((id, expense) => dispatch(startEditExpense(id, expense)))
  const startRemoveExpenseAction = useCallback(({ id }) => dispatch(startRemoveExpense({ id })))

  // fetch expense item from store
  const expenseItem = useSelector((state) => state.expenses.find((expense) => expense.id === props.match.params.id))

  // remove expense confirm modal
  const [removeExpenseDialogOpen, setRemoveExpenseDialogOpen] = useState(false)

  const onSubmit = (expense) => {
    startEditExpenseAction(expenseItem.id, expense)
    props.history.push('/')
  }
  const onConfirmRemoveExpense = () => {
    startRemoveExpenseAction({ id: expenseItem.id })
    setRemoveExpenseDialogOpen(false)
    props.history.push('/')
  }
  return (
    <div>
      <div className={classes.root}>
        <Paper elevation={0} square>
          <Container fixed>
            <Typography variant="h4" align="left">
              <Box fontWeight={300} m={1}>
                Edit Expense
              </Box>
            </Typography>
          </Container>
        </Paper>
      </div>
      <ExpenseForm2
        expense={expenseItem}
        onSubmit={onSubmit}
      />
      <div className='form-footer'>
        <Container fixed>
          <Grid
            container
            direction="column"
            spacing={1}
          >
            <Grid item xs={12}>
              <Divider />
            </Grid>
            <Grid container>
              <Grid container direction="row" alignItems="center" justify="flex-start" >
                <Grid item>
                  <ListItemIcon className='gear-icon'>
                    <SettingsIcon />
                  </ListItemIcon>
                </Grid>
                <Grid item>
                  <Typography variant="h6" align="left">
                    <Box fontWeight={300} m={1}>
                      Manage expense
                  </Box>
                  </Typography>
                </Grid>
              </Grid>
            </Grid>
            <Grid item xs={12}>
              <Button
                size="medium"
                variant="contained"
                color="secondary"
                startIcon={<DeleteIcon />}
                onClick={() => setRemoveExpenseDialogOpen(true)}
              >
                Remove Expense
            </Button>
              <RemoveExpenseDialog
                expense={expenseItem}
                removeExpenseDialogOpen={removeExpenseDialogOpen}
                setRemoveExpenseDialogOpen={setRemoveExpenseDialogOpen}
                onConfirmRemoveExpense={onConfirmRemoveExpense}
              />
            </Grid>
          </Grid>
        </Container>
      </div>
    </div>
  )
}

export { EditExpensePage as default }