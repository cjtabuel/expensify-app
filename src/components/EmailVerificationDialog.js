import React, { useState, useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux'
import { startLogout, sendEmailVerificationToUser } from '../actions/auth'

// MaterialUI imports
import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Grid,
  ListItemIcon,
  Snackbar,
  Typography,
} from '@material-ui/core'
import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
import MuiAlert from '@material-ui/lab/Alert';

const Alert = (props) => {
  return <MuiAlert elevation={6} variant="filled" {...props} />;
}

const EmailVerificationDialog = () => {
  const dispatch = useDispatch()
  const startLogoutAction = useCallback(() => dispatch(startLogout()))
  const startSendEmailVerificationToUserAction = useCallback(() => dispatch(sendEmailVerificationToUser()))

  const userEmail = useSelector(state => state.auth.userEmail)

  const [sendEmailVerificationSuccess, setSendEmailVerificationSuccess] = useState(false)
  const [sendEmailVerificationRequestError, setSendEmailVerificationRequestError] = useState(false)

  const handleSendVerification = () => {
    startSendEmailVerificationToUserAction()
      .then(() => {
        console.log('re-sending email')
        setSendEmailVerificationSuccess(true)
      })
      .catch(error => {
        const errCode = error.code

        if (errCode === "auth/too-many-requests") {
          setSendEmailVerificationRequestError(true)
          console.log(error)
        }
      })

  }
  const handleUserLogin = () => {
    console.log('going back to login')
    startLogoutAction()
  }
  const handleSaveEmailVerificationSuccess = (event, reason) => {
    if (reason === 'clickaway') {
      return
    }
    setSendEmailVerificationSuccess(false)
  }
  const handleSendEmailVerificationRequestError = (event, reason) => {
    if (reason === 'clickaway') {
      return
    }
    setSendEmailVerificationRequestError(false)
  }
  return (
    <div>
      <Dialog open={true} aria-labelledby="form-dialog-title">
        <DialogTitle id="form-dialog-title" component='div'>
          <Grid container direction='row' justify="flex-start" alignItems="center">
            <Grid item>
              <ListItemIcon fontSize='large'>
                <CheckCircleOutlineIcon color='primary' fontSize='large' />
              </ListItemIcon>
            </Grid>
            <Grid item>
              <Typography variant="h5" align="left" component='div'>
                <Box fontWeight={300}>
                  Verify your email
                </Box>
              </Typography>
            </Grid>
          </Grid>
        </DialogTitle>
        <DialogContent dividers component='div'>
          <DialogContentText component='div'>
            <Typography variant="h6" component='div'>
              <Box fontWeight={500} m={1} >
                Thank you for signing up for Expensify!
              </Box>
            </Typography>
          </DialogContentText>
          <DialogContentText component='div'>
            <Typography component='div'>
              <Box fontWeight={300} m={1}>
                We want to make sure the credentials you are using belong to you. We have sent a verification email to {userEmail} for you to verify your account.
              </Box>
            </Typography>
          </DialogContentText>
          <DialogContentText component='div'>
            <Typography component='div'>
              <Box fontWeight={300} m={1}>
                Once you have been verified, you may proceed to the login page of the Expensify Application.
              </Box>
            </Typography>
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleSendVerification} color='secondary' variant='contained'>
            Re-send Verification
          </Button>
          <Button onClick={handleUserLogin} color="primary" variant='contained'>
            Login
          </Button>
        </DialogActions>
      </Dialog>
      <Snackbar
        open={sendEmailVerificationSuccess}
        autoHideDuration={6000}
        onClose={handleSaveEmailVerificationSuccess}
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
      >
        <Alert onClose={handleSaveEmailVerificationSuccess} severity="success">
          A verification email has been to {userEmail}
        </Alert>
      </Snackbar>
      <Snackbar
        open={sendEmailVerificationRequestError}
        autoHideDuration={6000}
        onClose={handleSendEmailVerificationRequestError}
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
      >
        <Alert onClose={handleSendEmailVerificationRequestError} severity="error">
          Too many requests have been made from this device. Please try again later.
        </Alert>
      </Snackbar>
    </div>
  );
}

export default EmailVerificationDialog