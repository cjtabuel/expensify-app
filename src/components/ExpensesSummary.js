import React from 'react'
import { connect } from 'react-redux'
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

const ExpensesSummary = (props) => (
  <div>
    <p>
      Viewing {props.expenseCount} totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
    </p>
  </div>
)

const mapStateToProps = (state) => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
  }
}

export { ExpensesSummary }
export default connect(mapStateToProps)(ExpensesSummary)