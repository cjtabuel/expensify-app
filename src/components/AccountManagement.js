import React, { useState, useCallback } from 'react'
import { useDispatch } from 'react-redux'

import AccountManagementDialog from './AccountManagementDialog'
import { deleteUserAccount } from '../actions/auth'

import {
  Button,
  Box,
  Grid,
  ListItemIcon,
  Typography
} from '@material-ui/core'

import SettingsIcon from '@material-ui/icons/Settings';
import WarningIcon from '@material-ui/icons/Warning';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    '& > *': {
      marginBottom: theme.spacing(6),
    }
  },
}))

const AccountManagement = () => {
  const classes = useStyles()
  const dispatch = useDispatch()

  const [accountManagementDialogOpen, setAccountManagementDialogOpen] = useState(false)

  const deleteUserAccountAction = useCallback(() => dispatch(deleteUserAccount()))

  const handleDeleteUserAccount = () => {
    deleteUserAccountAction()
  }
  return (
    <div className={classes.root}>
      <Grid container spacing={2}>
        <Grid item xs={12}>
          <Grid container direction="row" alignItems="center" justify="flex-start">
            <Grid item xs={1}>
              <ListItemIcon>
                <SettingsIcon />
              </ListItemIcon>
            </Grid>
            <Grid item>
              <Typography variant="h6" align="left">
                <Box fontWeight={300} m={1}>
                  Account Management
                </Box>
              </Typography>
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <Button
            size="medium"
            variant="contained"
            color="secondary"
            startIcon={<WarningIcon />}
            endIcon={<WarningIcon />}
            onClick={() => setAccountManagementDialogOpen(true)}
          >
            Delete Account
          </Button>
        </Grid>
      </Grid>
      <AccountManagementDialog
        accountManagementDialogOpen={accountManagementDialogOpen}
        setAccountManagementDialogOpen={setAccountManagementDialogOpen}
        handleDeleteUserAccount={handleDeleteUserAccount}
      />
    </div>
  )
}

export default AccountManagement