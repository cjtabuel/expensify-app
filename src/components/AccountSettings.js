import React from 'react'
import { useSelector } from 'react-redux'
import UpdatePasswordForm from './UpdatePasswordForm'
import AccountManagement from './AccountManagement'
import UpdateEmailForm from './UpdateEmailForm'
import CurrencySetting from './CurrencySetting'

import {
  Box,
  Container,
  Grid,
  Typography
} from '@material-ui/core'

const AccountSettings = () => {
  const loginMethod = useSelector((state) => state.auth.loginMethod)

  return (
    <div>
      <Container fixed>
        <Grid container spacing={0} alignItems="center" justify='center'>
          <Grid item xs={12}>
            <Typography variant="h5" align="left">
              <Box fontWeight={300} m={1}>
                Account Settings
              </Box>
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <hr />
          </Grid>
        </Grid>

        <CurrencySetting />
        {loginMethod === 'password' && <UpdateEmailForm />}
        {loginMethod === 'password' && <UpdatePasswordForm />}
        <AccountManagement />
      </Container>
    </div>
  )
}



export default AccountSettings