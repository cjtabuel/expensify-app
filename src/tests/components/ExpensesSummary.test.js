import React from 'react'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import { shallow } from 'enzyme'

test('should render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={250} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={750} />)
  expect(wrapper).toMatchSnapshot()
})

