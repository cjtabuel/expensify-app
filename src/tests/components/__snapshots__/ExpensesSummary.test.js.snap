// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.50
  </p>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totalling 
    $7.50
  </p>
</div>
`;
